stl/inc/format

```diff
@@ -2514,32 +3109,142 @@ _NODISCARD _OutputIt _Fmt_write(
+    }
+
+    constexpr _Measure_string_prefix_iterator_utf& operator++() noexcept {
+        ++_WrappedIter;
+        return *this;
+    }
+};
+
+
+class _Measure_string_prefix_iterator_legacy {
+    _Fmt_codec<char, false> _Codec;
+    const char* _First = nullptr;
+    const char* _Last  = nullptr;
+    int _Units         = 0;
+
+    constexpr friend bool operator==(
+        const _Measure_string_prefix_iterator_legacy& _Lhs, _Measure_string_prefix_end_iterator) noexcept {
+        return _Lhs._First >= _Lhs._Last;
+    }
+    constexpr friend bool operator==(const _Measure_string_prefix_iterator_legacy& _Lhs,
+        const _Measure_string_prefix_iterator_legacy& _Rhs) noexcept {
+        return _Lhs._First == _Rhs._First && _Lhs._Last == _Rhs._Last;
+    }
+
+    void _Update_units() {
+        if (_First < _Last) {
+            _Units = _Codec._Units_in_next_character(_First, _Last);
+        } else {
+            _Units = -1;
+        }
+    }
+
+public:
+    using iterator_concept = forward_iterator_tag;
+    using difference_type  = ptrdiff_t;
+    using value_type       = int;
+
+    _Measure_string_prefix_iterator_legacy(const char* _First_val, const char* _Last_val)
+        : _First(_First_val), _Last(_Last_val) {
+        _Update_units();
+    }
+
+    _Measure_string_prefix_iterator_legacy() = default;
+
+    _Measure_string_prefix_iterator_legacy& operator++() noexcept {
+        _First += _Units;
+        _Update_units();
+        return *this;
+    }
+    _Measure_string_prefix_iterator_legacy operator++(int) noexcept {
+        auto _Old = *this;
+        ++*this;
+        return _Old;
+    }
+    value_type operator*() const {
+        return _Units;
+    }
+
+    const char* _Position() const {
+        return _First;
+    }
+};
+
+template <class _CharT, bool _Statically_Utf8 = _Is_execution_charset_utf8()>
+struct _Measure_string_prefix_iterator : _Measure_string_prefix_iterator_utf<_CharT> {
+    using _Measure_string_prefix_iterator_utf<_CharT>::_Measure_string_prefix_iterator_utf;
+};
+
+template <>
+struct _Measure_string_prefix_iterator<char, false> : _Measure_string_prefix_iterator_legacy {
+    using _Measure_string_prefix_iterator_legacy::_Measure_string_prefix_iterator_legacy;
+};
+
 template <class _CharT>
 _NODISCARD const _CharT* _Measure_string_prefix(const basic_string_view<_CharT> _Value, int& _Width) {
     // Returns a pointer past-the-end of the largest prefix of _Value that fits in _Width, or all
     // of _Value if _Width is negative. Updates _Width to the estimated width of that prefix.
     const int _Max_width = _Width;
-    auto _Pos            = _Value.data();
-    const auto _Last     = _Pos + _Value.size();
-    int _Estimated_width = 0; // the estimated width of [_Value.data(), _Pos)
-    const _Fmt_codec<_CharT> _Codec;
+    _Measure_string_prefix_iterator<_CharT> _Pfx_iter(_Value.data(), _Value.data() + _Value.size());
+    auto _Pos               = _Value.data();

```

Comment:
`_Pos` was originally updated, which is why it's non-`const`. Now that logic has been moved into `_Pfx_iter`, if I understand this correctly, so `_Pos` is never modified (and is used only to compute `_Last` which is returned). I suspect that:
* `_Pos` should become `const`.
* `_Pos` should be renamed to `_First`.
* The comment `// the estimated width of [_Value.data(), _Pos)` should be updated.